api.js

```diff
@@ -61,19 +62,44 @@ Api.prototype._reset = function () {
 	this.base = '';
 };
 
-Api.prototype._runFile = function (file) {
+Api.prototype._runFile = function (file, onForkStarting) {
 	var options = objectAssign({}, this.options, {
 		precompiled: this.precompiler.generateHashForFile(file)
 	});
 
-	return fork(file, options)
-		.on('teardown', this._handleTeardown)
-		.on('stats', this._handleStats)
-		.on('test', this._handleTest)
-		.on('unhandledRejections', this._handleRejections)
-		.on('uncaughtException', this._handleExceptions)
-		.on('stdout', this._handleOutput.bind(this, 'stdout'))
-		.on('stderr', this._handleOutput.bind(this, 'stderr'));
+	var execArgv = process.execArgv.slice();
+	var debugArgIndex = -1;
+	for (var i = 0; i < execArgv.length; i++) {
+		if (execArgv[i].indexOf('--debug-brk=') === 0 || execArgv[i].indexOf('--debug=') === 0) {
```

Comment:
Because we don't append flag, but change existing. By default `child_process.fork` pass `process.execArgv` to child process as is. What we do here — we find debug argument and change its value to any free port.